[Docs Review] Review of #1115 - First version of migration docs (#1131)
pokt-network/poktroll: docusaurus/docs/operate/morse_migration/claiming.md

| **Morse Account** (Unstaked)   | ✅ Unstaked balance 1:1; E.g. 1000 uPOKT in Morse → 1000 uPOKT in Shannon                   | ❌ N/A                                                                            | ❌ N/A                                                                                                         |
| **Morse Application** (Staked) | ✅ Unstaked balance 1:1; E.g. 500 uPOKT unstaked in Morse → 500 uPOKT unstaked in Shannon  | ✅ App stake 1:1 E.g. 15000 uPOKT staked in Morse → 15000 uPOKT staked in Shannon | ❌ N/A                                                                                                         |
| **Morse Servicer**             | ✅ Unstaked balance 1:1; E.g. 2000 uPOKT unstaked in Morse → 2000 uPOKT unstaked in Shannon | ❌ N/A                                                                            | ✅ Servicer stake 1:1 E.g. 15000 uPOKT staked as Servicer in Morse → 15000 uPOKT staked as Supplier in Shannon |

### Summary - Important Notes

1. **Account Claiming:**

>    - A basic Morse account (unstaked) can only claim as a Shannon account

Reviewer: ```suggestion
   - A basic Morse account (unstaked) MAY ONLY claim as a Shannon account
```

   - Only the unstaked balance transfers over (1:1 ratio)

2. **Application Claiming:**

>    - A Morse Application must claim as a Shannon Application

Reviewer: ```suggestion
   - A Morse Application MAY ONLY claim as a Shannon Application
```

   - Both unstaked balance and application stake transfer over (1:1 ratio)

>    - While stake amounts remain the same, Shannon allows for modified staking configurations

Reviewer: ```suggestion
   - While stake amounts remain the same, a Shannon service ID is required and must be provided at the time of claiming.
```

3. **Servicer to Supplier Claiming:**

>    - A Morse Servicer must claim as a Shannon Supplier

Reviewer: ```suggestion
   - A Morse Servicer MAY ONLY claim as a Shannon Supplier
```

   - Both unstaked balance and staked amount transfer over (1:1 ratio)

>    - The stake amount is automatically determined from the Morse account's supplier stake amount

Reviewer: ```suggestion
   - While stake amounts remain the same, a Shannon supplier stake configuration is required and must be provided at the time of claiming
```

## Morse & Shannon Keyrings

:::tip Cosmos SDK Keyring

You can learn more about Cosmos SDK Keyrings [here](https://docs.cosmos.network/v0.46/run-node/keyring.html)

:::

A **keyring** holds the private/public keypairs used to interact with a node.

Both `Morse` and `Shannon` have their own **COMPLETELY INDEPENDENT** keyrings which they use to manage their respective account keys.

**ALL claim operations** require the use of a private key from **BOTH** of these keyrings:

- The `Morse` key is used to generate the `morse_signature` field which is common to all claim messages.
- The `Shannon` key is used to sign the `MsgClaimMorse...` message which is broadcast to the Shannon network.

Since the Shannon CLI (`poktrolld`) is used to broadcast the `ClaimTx`, the prerequisites are:

> 1. Export the `Morse` key (securely) from your `pocket` (i.e. Morse CLI) so that it can be used by `poktrolld` (i.e. Shannon CLI).

Reviewer: ```suggestion
1. Export the `Morse` key (securely) from `pocket` (i.e. Morse CLI) so that it can be used by `poktrolld` (i.e. Shannon CLI).
```

2. Ensure that the desired `Shannon` destination account key exists in the `Shannon` keyring.

### Morse Keyring CLI - Exporting Existing Account Keys

In order to claim any Morse (staked or not) account on Shannon, the corresponding private key MUST be exported from the `pocket` CLI such that it can be used by `poktrolld` CLI.

It is necessary to sign the respective claim message's `morse_signature` field. This cryptographic primitive is critical as it establishes the authenticity of the claim.

> The Morse CLI provides a subcommand which exports a given Morse account private key as an encrypted file `pocket accounts export <morse_address>`.

Reviewer: ```suggestion
The Morse CLI provides a subcommand which exports a given Morse account private key as an encrypted file: `pocket accounts export <morse_address>`.
```

```

:::important

You can replace the key name (i.e. `examplekey` in the example above) with any name you choose.

The key name (not address) will be required later when claiming the account to indicate the Shannon destination account.
:::

## Claiming Morse Accounts

### Migration Module Tx CLI: `poktrolld tx migrate`

The `poktrolld` CLI provides a set of subcommands which can be used to claim Morse accounts on Shannon.

These subcommands map 1:1 to the different claim messages (see image above) which are used, depending on whether the Morse account being claimed is staked, and if so, as what actor type.

You can run the following command to see all available subcommands:

```shell
poktrolld tx migrate --help
```

Which should produce output similar to the following:

```shell
  claim-account
  claim-application
  claim-supplier

>   collect-morse-accounts Collect account balances and stakes from [morse-state-export-path] JSON file and output to [morse-account-state-path] as JSON

Reviewer: ```suggestion
  collect-morse-accounts Collect account balances and stakes from [morse-state-export-path] JSON file and output to [morse-account-state-path] as JSON
  import-morse-accounts
  validate-morse-accounts
```

:::

### Claim a Basic Morse Account (Unstaked, Non-Actor)

:::note

If the Morse account is not staked as an Application or Supplier, it is considered an "unstaked" or "non-actor" account.

:::

Claiming an unstaked account will mint the unstaked balance of the Morse account being claimed to the Shannon account which the signer of the `MsgClaimMorseAccount`.

This unstaked balance amount is retrieved from the corresponding onchain `MorseClaimableAccount` which was imported by the foundation.

For example, running the following command:

```bash
poktrolld migrate claim-account \
  ./pocket-account-8b257c7f4e884e49bafc540d874f33f91436e1dc.json \
  --from app1
```

Should prompt for a passphrase and produce output similar to the following:

```shell
Enter Decrypt Passphrase:
MsgClaimMorseAccount {
  "shannon_dest_address": "pokt1mrqt5f7qh8uxs27cjm9t7v9e74a9vvdnq5jva4",
  "morse_src_address": "8B257C7F4E884E49BAFC540D874F33F91436E1DC",
  "morse_signature": "hLGhLRjP6jgP6wgOIaYFxIxT3z4jb4IBDKovMkX5AqUsOqdF+rEIO5aofOKnmYW9BkqL0v2DfUfE3nj25FNhBA=="
}
Confirm MsgClaimMorseAccount: y/[n]:
```

> ### Claim a Morse Application (Staked, Actor)

Reviewer: Why change this H3 and not the following one?

> ### Claim a Supplier (Servicer) Staked Account

Author: I don't fully understand this.

Update the ToC and here is where things stand now:

![Screenshot 2025-03-18 at 2 00 17 PM](https://github.com/user-attachments/assets/a54220ef-29d6-4856-8661-6ef2432bb325)

  "morse_src_address": "8B257C7F4E884E49BAFC540D874F33F91436E1DC",
  "morse_signature": "hLGhLRjP6jgP6wgOIaYFxIxT3z4jb4IBDKovMkX5AqUsOqdF+rEIO5aofOKnmYW9BkqL0v2DfUfE3nj25FNhBA=="
}
Confirm MsgClaimMorseAccount: y/[n]:
```

:::tip

See `poktrolld tx migrate claim-application --help` for more details.

:::

### Claim a Supplier (Servicer) Staked Account

:::warning TODO_MAINNET(@bryanchriswhite, #1126)

Update with respect to owner/operator custodia/non-custodial addresses.

:::

#### Critical Analogy: Morse Servicer to Shannon Supplier

_tl;dr Claiming a Morse Servicer as a Shannon Supplier is like staking a new Shannon Supplier without explicitly specifying the `stake_amount`_

Claiming a Morse supplier requires a supplier config that is identical to that used by the Shannon CLI when staking a supplier (as opposed to claiming). See the [supplier staking config](../configs/supplier_staking_config.md).

The only difference between staking a new supplier on Shannon and **claiming** an existing supplier from Morse to Shannon is that the `stake_amount` is determined by the corresponding onchain `MorseClaimableAccount`'s supplier stake amount.

:::important Omit `stake_amount`

> Omit the `stake_amount` field in a supplier config; providing it in when claiming a Morse supplier is an error.\*\*

Reviewer: ```suggestion
Omit the `stake_amount` field in a supplier config; providing it in when claiming a Morse supplier is an error.
```
